samplecode/java/com/athenahealth/api/APIConnection.java

```diff
@@ -356,8 +354,8 @@ private String path_join(String ... args) {
 	 * @param parameters keys and values to encode
 	 * @return the query string
 	 */
-	private String urlencode(Map<?, ?> parameters) {
-		StringBuilder sb = new StringBuilder();
+	private String encodeUrl(Map<?, ?> parameters) {
```

Comment:
I think this is one of two changes I don't like. urlencode is a wrapper around URLEncoder.encode, so changing the name to encodeUrl is not adding any value. urlEncode should be sufficient.

Reply:
`URLEncoder` is a noun and perfect for a class. But there is something odd about the name `urlencode` for method. Putting aside the fact that it doesn't follow camelCase, it's not a proper verb. I have used a lot of good, fluent API's and they all tell what the method does clearly. `encodeUrl` says "encodes the url", it's straightforward. `urlEncode` does this, but adds an additional overhead on the readability. For instance, JS has such an API. It has `encodeURI` and `encodeURIComponent` methods. If the same thing were `uriEncode` or uriComponentEncode`, it's odd to read.

So though changing the name to `encodeUrl` seems like not adding any value, it certainly does by improving the readability.

Reply:
ok